sources/web/datalab/auth.ts

```diff
@@ -145,24 +169,22 @@ export function handleAuthFlow(request: http.ServerRequest, response: http.Serve
         }
       });
     }
-  } else if (!oauth2Client && process.env.DATALAB_ENV == 'local' && !fs.existsSync(appCredFile)) {
+  } else if (!oauth2Client && !isSignedIn()) {
```

Comment:
Add "path.indexof('/signin') == 0" here, in case the function is called with another path?